### DISCUSSION TITLE
SkyTree's README.md is kinda lacking in explainations.

### DISCUSSION THREAD

**Discussion by l1m3r** [redseiko/ComfyMods#69] [Q&A] [answered]
I like the mod - thank you for developing it - but I don't want to run all the tests myself to figure out what works or doesn't and how:

1. is the mod required on a dedicated server?
2. is Yggdrasil changed permanently once a map has been generated and afterwards loaded once with SkyTree?
3. Same as 2.: Does SkyTree need to stay active for the tree to stay solid or is that a one time operation? (if not one time, could that maybe be added to UpgradeWorld (just a thought)?)
4. related to 1. - 3.: Does every client need the mod or will stuff on the tree just break if someone without the mod runs the respective chunk?
5. What exactly does "_Mod should be re-enabled once zone/sector generation has occurred._" mean? Does the whole world need to generated before enabling SkyTree or is the normal way of just _starting_ a new world (then enabling SkyTree) and letting the normal world generation happen through discovery by players enough?

I'll gladly help testing/answering those questions if necessary but the current [README.md](https://github.com/redseiko/ComfyMods/tree/main/SkyTree#readme) doesn't really answer them yet. :-/

**Comment by redseiko** [comment, marked as answer]
1) No.
2) No.
3) Yes. No idea about UpgradeWorld.
4) All clients need mod.
5) Generate the whole world with mod disabled (fly through areas where tree overlaps). Will probably fix this in an update though it's complicated to do it properly.

README is written on a best effort basis with the free time I have to create and maintain all the mods I release to the public.

**Reply by l1m3r** [reply]
Thank you very much for the quick reply.
I assumed correctly on most questions but I hoped differently (permanently solid Yggdrasil). ;-)

5. That is a huge inconvenience.

How difficult would it be to get a 2D projection of Yggdrasil onto the worlds surface (coordinates)?
[UpgradeWorld](https://github.com/JereKuusela/valheim-upgrade_world#readme) has the ability to generate a whole world in one go (`zones_generate`) but it takes hours and when Ashlands/DeepNorth(/Oceans?) get released one would need to manually update those zones again.
But It can also generate zones in circles with arbitrary diameter and center coordinates. -> 2D projection of Yggdrasil + some math + UpgradeWorld = it would be possible to only pre-generate all zones under the tree.

Shall I suggest that as a new feature to [JereKuusela](https://github.com/JereKuusela) - `zones_generate Yggdrasil`?

**Reply by redseiko** [reply]
I'll push out an update to modify the prefab layers back to skybox during a SpawnZone() call and then back to static_solid afterwards.

No idea on the math.

**Comment by l1m3r** [comment]
> modify the prefab layers back to skybox during a SpawnZone() call and then back to static_solid afterwards.

yeah - I was too quick with my suggestion. Just thought about
> Will probably fix this in an update though it's complicated to do it properly.

and
> Sets the layer for the YggdrasilBranch prefab/children from 19 (skybox) to 15 (static_solid).

again right after my last reply and came to the _same idea_ (in principle not actual code/functions/whatnot) you apparently already had. :-)

It's better than pre-generating zones anyway.
